haskell/experimental/private/module.bzl

```
@@ -188,6 +188,11 @@ def _build_haskell_module(
 
     args.add_all(cc.include_args)
 
+    if plugins or enable_th:
+        # cc toolchain linker flags would be necessary when the interpreter wants to
+        # load any libraries
+        args.add_all(["-optl" + f for f in cc.linker_flags])
```

```suggestion
        args.add_all(cc.linker_flags, format_each = "-optl%s")
```
See [optimization recommendations](https://docs.bazel.build/versions/main/skylark/performance.html#use-ctxactionsargs-for-command-lines).

---

Done.